airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -111,4 +125,105 @@ private WbConnectionRead buildWbConnectionRead(ConnectionRead connectionRead) th
     return wbConnectionRead;
   }
 
+  public WbConnectionRead webBackendGetConnection(WebBackendConnectionIdRequestBody webBackendConnectionIdRequestBody)
+      throws ConfigNotFoundException, IOException, JsonValidationException {
+    final ConnectionIdRequestBody connectionIdRequestBody = new ConnectionIdRequestBody()
+        .connectionId(webBackendConnectionIdRequestBody.getConnectionId());
+
+    final ConnectionRead connection = connectionsHandler.getConnection(connectionIdRequestBody);
+
+    if (webBackendConnectionIdRequestBody.getWithRefreshedCatalog() != null && webBackendConnectionIdRequestBody.getWithRefreshedCatalog()) {
+      final SourceIdRequestBody sourceId = new SourceIdRequestBody().sourceId(connection.getSourceId());
+      final SourceDiscoverSchemaRead discoverSchema = schedulerHandler.discoverSchemaForSourceFromSourceId(sourceId);
+
+      final SourceSchema original = connection.getSyncSchema();
+      final SourceSchema discovered = discoverSchema.getSchema();
+      final SourceSchema combined = updateSchemaWithDiscovery(original, discovered);
+
+      connection.setSyncSchema(combined);
+    }
+
+    return buildWbConnectionRead(connection);
+  }
+
+  @VisibleForTesting
+  protected static SourceSchema updateSchemaWithDiscovery(SourceSchema original, SourceSchema discovered) {
+    Map<String, SourceSchemaStream> originalStreamsByName = original.getStreams().stream()
+        .collect(toMap(SourceSchemaStream::getName, s -> s));
+
+    List<SourceSchemaStream> streams = new ArrayList<>();
+
+    for (SourceSchemaStream stream : discovered.getStreams()) {
+      SourceSchemaStream originalStream = originalStreamsByName.get(stream.getName());
+
+      if (originalStream != null) {
+        Set<String> fieldNames = stream.getFields().stream().map(SourceSchemaField::getName).collect(toSet());
+        stream.setSelected(originalStream.getSelected());
+
+        if (stream.getSupportedSyncModes().contains(originalStream.getSyncMode())) {
+          stream.setSyncMode(originalStream.getSyncMode());
+        }
+
+        Set<String> updatedCursorFields = new HashSet<>();
+
+        for (String oldCursorField : originalStream.getCursorField()) {
+          if (fieldNames.contains(oldCursorField)) {
```

Comment:
i think there may be a misunderstanding here. a cursor field is an array because it is a breadcrumb into an object. e.g. `{ "a": { "b" : 1 } }`. For example in this object if the cursor field were b, the cursorField array would be `["a", "b"]`. 

This does _not_ line up well with the model in the API right now, because the API model assumes everything is flat ( table / column shaped). Pretty much anything we do here is going to be a bit hacky, since the data models don't line up well. What I would suggest is `fieldNames.contains(String.join(".", originalStream.getCursorField());)` or something similar. lmk if this still doesn't make sense. it is weird.